service/src/main/java/io/camunda/service/license/CamundaLicense.java

```diff
@@ -32,39 +32,45 @@ public synchronized boolean isValid() {
     return isValid;
   }
 
-  public synchronized boolean isSelfManaged() {
-    return isSelfManaged;
+  public synchronized String getLicenseType() {
+    return licenseType;
   }
 
   public synchronized void initializeWithLicense(final String license) {
     if (!isInitialized) {
       isValid = determineLicenseValidity(license);
-      isSelfManaged = determineIfLicenseEnvModeIsSelfManaged(license);
+      licenseType = getLicenseTypeFromProperty(license);
 
       isInitialized = true;
     }
   }
 
   /**
-   * SaaS mode is determined through the properties of the passed in license.
+   * Camunda licenses will have a license type property, fetch that out of the license and return
+   * the value
    *
    * <p>Self-managed mode is any other possibility. (ex, blank license, prop missing, etc)
    */
-  private boolean determineIfLicenseEnvModeIsSelfManaged(final String licenseStr) {
+  private String getLicenseTypeFromProperty(final String licenseStr) {
```

Comment:
🔧 We can improve a bit this method, this is a suggestion but there are other ways to do it 😄 

```
try {
    final LicenseKey licenseKey = getLicenseKey(licenseStr);
    String licenseType = licenseKey.getProperties().getOrDefault(LICENSE_TYPE_PROPERTY_NAME, UNKNOWN_LICENSE_TYPE);

    if (UNKNOWN_LICENSE_TYPE.equals(licenseType)) {
        LOGGER.error("Expected a licenseType property on the Camunda License, but none were found.");
    }

    return licenseType;

} catch (InvalidLicenseException e) {
    LOGGER.error("Encountered an invalid license when determining the license type.", e);
} catch (Exception e) {
    LOGGER.error("Unexpected error while determining the license type.", e);
}
return UNKNOWN_LICENSE_TYPE;
```